sisl/grid.py

```diff
@@ -179,6 +179,84 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
+            the shape of the final output. If not provided, the current shape of the grid
+            will be used. 
+            
+            Notice however that if the transformation applies a big shear to the image (grid)
+            you will probably need to have a bigger output_shape.
+        mode: str, optional
+            determines how to handle borders. See scipy docs for more info on the possible values.
+        order : int 0-5, optional
+            the order of the spline interpolation to calculate the values (since we are applying
+            a transformation, we don't actually have values for the new locations and we need to
+            interpolate them)
+            1 means linear, 2 quadratic, etc...
+        **kwargs:
+            the rest of keyword arguments are passed directly to `scipy.ndimage.affine_transform`
+
+        See also
+        ----------
+        scipy.ndimage.affine_transform : method used to apply the linear transformation.
+        """
+        # Take the current shape of the grid if no output shape was provided
+        if output_shape is None:
+            output_shape = self.shape
+
+        # Get the current cell in coordinates of the destination axes
+        inv_axes = np.linalg.inv(axes)
+        projected_cell = self.cell.dot(inv_axes)
+
+        # From that, infere how long will the bounding box of the cell be
+        lengths = abs(projected_cell).sum(axis=0)
+
+        # Create the transformation matrix. Since we want to control the shape
+        # of the output, we can not use self.dcell directly, we need to modify it.
+        scales = output_shape / lengths
+        forward_t = (self.dcell.dot(inv_axes)*scales).T
+
+        # Scipy's affine transform asks for the inverse transformation matrix, to
+        # map from output pixels to input pixels. By taking the inverse of our
+        # transformation matrix, we get exactly that.
+        tr = np.linalg.inv(forward_t)
+
+        # Calculate the offset of the image so that all points of the grid "fall" inside
+        # the output array. 
+        # For this we just calculate the centers of the input and output images
+        center_input = 0.5 * (np.array(self.shape) - 1)
+        center_output = 0.5 * (np.array(output_shape) - 1)
+
+        # And then make sure that the input center that is interpolated from the output
+        # falls in the actual input's center
+        offset = center_input - np.dot(tr, center_output)
+
+        # We pass all the parameters to scipy's affine_transform
+        transformed_image = affine_transform(self.grid, tr, order=1, offset=offset, 
+            output_shape=output_shape, mode=mode, **kwargs)
+
+        # Create a new grid with the new shape and the new axes (notice how the axes
+        # are rescaled from the input axes to fit the actual coordinates of the system)
+        grid = self.__class__(output_shape, sc=axes*lengths.reshape(3,1))
```

Comment:
here it may make sense to make the `self.geometry` match the transformed shape and linear vectors?
In that way we have atomic coordinates again.

Reply:
Also, just pass `[1, 1, 1]` as the shape, when overriding the `grid` in the next line the correct shape will be used.
This will reduce memory requirements by a half!